tests/testthat/test-qenv_extract.R

```diff
@@ -0,0 +1,68 @@
+
+testthat::test_that("`[.` returns empty qenv for names not in qenv", {
+  data <- within(qenv(), {
+    x <- 1
+    a <- 2
+  })
+  testthat::expect_warning(
+    testthat::expect_equal(data["y"], qenv()),
+    "None of 'names' elements exist in 'qenv'. Returning empty 'qenv'."
+  )
+})
+
+testthat::test_that("`[.` returns limited qenv for some names not in qenv", {
+  data <- within(qenv(), {
+    x <- 1
+    a <- 2
+  })
+  testthat::expect_warning(
+    testthat::expect_equal(data[c("y", "a")], data["a"]),
+    "Some elements of 'names' do not exist in 'qenv'. Skipping those: y."
+  )
+})
+
+testthat::test_that("`[.` extracts proper objects", {
+  q <- qenv()
+  code <- c("x<-1", "a<-1;b<-2")
+  q <- eval_code(q, code)
+  object_names <- c("x", "a")
+  qs <- q[object_names]
+  testthat::expect_true(all(ls(get_env(qs)) %in% object_names))
+})
+
+testthat::test_that("`[.` extract proper code", {
+  q <- qenv()
+  code <- c("x<-1", "a<-1;b<-2")
+  q <- eval_code(q, code)
+  object_names <- c("x", "a")
+  qs <- q[object_names]
+  testthat::expect_identical(
+    unlist(qs@code),
+    c("x<-1\n", "a<-1;")
+  )
+})
+
+testthat::test_that("`[.` preservers comments in the code", {
```

Comment:
There should be multiple comment-preserving-tests

- Comments are associated with the call from the same line (within same eval_code)
- Comments are associated with following call (within same eval_code)
- Comments in the end of the src are associated with the last call (within same eval_code)
- Comments alone (without code) are added as a separate code element

```suggestion
testthat::test_that("`[.` comments are preserved in the code and associated with the following call", {
```

Reply:
Hey, will continue on this one now

Reply:
Hey @gogonzo we had few tests for comments, but in other file. I moved them and extended them. They now leave in eval_code e28b937bdc3e927c576de4d86eb163d26dbf17ed